projects/sudoers/build.sh

```diff
@@ -15,25 +15,70 @@
 #
 ################################################################################
 
-if [ $SANITIZER == "address" ]; then
-    export LDFLAGS="-fsanitize=address"
-elif [ $SANITIZER == "undefined" ]; then
-    export LDFLAGS="-fsanitize=undefined"
-elif [ $SANITIZER == "coverage" ]; then
-    export LDFLAGS="$CFLAGS"
+# Debugging
+env
+
+# Move ASAN-specific flags into ASAN_CFLAGS and ASAN_LDFLAGS
+# That way they don't affect configure but will get used when building.
+if [ $SANITIZER == "coverage" ]; then
+    export ASAN_CFLAGS="$COVERAGE_FLAGS"
+    export ASAN_LDFLAGS="$COVERAGE_FLAGS"
+    CFLAGS="`echo \"$CFLAGS\" | sed \"s/ $COVERAGE_FLAGS//\"`"
+else
+    export ASAN_CFLAGS="$SANITIZER_FLAGS"
+    export ASAN_LDFLAGS="$SANITIZER_FLAGS"
+    CFLAGS="`echo \"$CFLAGS\" | sed \"s/ $SANITIZER_FLAGS//\"`"
 fi
 
-./configure --enable-static-sudoers --enable-static --disable-shared-libutil
-make
+# Build sudo with static libs for simpler fuzzing
+./configure --enable-static-sudoers --enable-static --disable-shared-libutil \
```

Comment:
I wonder if you could accomplish the same thing as the if-else with:
`CFLAGS= ./configure...` on this line. That would probably be much simpler.

Reply:
But then the sudo object files wouldn't be built with the sanitizer/coverage flags.  Sudo still needs to build with those flags. The problem is that they cause issues for the configure script itself which led to spurious failures due to mis-compilation (the return value of reallocarray() was truncated to an int).